hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsWebClientConfiguration.kt

```diff
@@ -61,10 +72,51 @@ class HmppsReactiveWebClientConfiguration {
   fun reactiveOAuth2AuthorizedClientManager(
     reactiveClientRegistrationRepository: ReactiveClientRegistrationRepository,
     reactiveOAuth2AuthorizedClientService: ReactiveOAuth2AuthorizedClientService,
+    reactiveOAuth2AuthorizedClientProvider: ReactiveOAuth2AuthorizedClientProvider,
   ): ReactiveOAuth2AuthorizedClientManager = AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(
     reactiveClientRegistrationRepository,
     reactiveOAuth2AuthorizedClientService,
-  ).kotlinApply { setAuthorizedClientProvider(ReactiveOAuth2AuthorizedClientProviderBuilder.builder().clientCredentials().build()) }
+  ).kotlinApply { setAuthorizedClientProvider(reactiveOAuth2AuthorizedClientProvider) }
+
+  @ConditionalOnMissingBean
+  @Bean
+  fun reactiveOAuth2AuthorizedClientProvider(
+    builder: WebClient.Builder,
+  ): ReactiveOAuth2AuthorizedClientProvider = builder.reactiveOAuth2AuthorizedClientProvider(Duration.ofSeconds(DEFAULT_TIMEOUT_SECONDS))
+}
+
+fun oAuth2AuthorizedClientProvider(clientCredentialsClientTimeout: Duration): OAuth2AuthorizedClientProvider {
+  // TODO We have to use a deprecated class here because that's what Spring Security uses. If Spring Security switches to RestClientClientCredentialsTokenResponseClient then we can do the same.
+  val accessTokenResponseClient = DefaultClientCredentialsTokenResponseClient().kotlinApply {
+    setRestOperations(
+      RestTemplate(listOf(FormHttpMessageConverter(), OAuth2AccessTokenResponseHttpMessageConverter())).kotlinApply {
+        errorHandler = OAuth2ErrorResponseErrorHandler()
+        requestFactory = SimpleClientHttpRequestFactory().kotlinApply {
+          setReadTimeout(clientCredentialsClientTimeout)
+        }
+      },
+    )
+  }
```

Comment:
Using the deprecated class is rubbish, but when I tried switching to the replacement class nothing worked - hence we need to wait for Spring Security to update